core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -869,18 +870,20 @@ object Pull extends PullLowPriority {
               case Some(s) => s.close(ExitCase.Succeeded).rethrow.as(true)
             }
           else F.pure(false)
-
-        val tail: F[R[G, X]] = maybeCloseExtendedScope.flatMap { closedExtendedScope =>
-          val newExtendedScope = if (closedExtendedScope) None else extendedTopLevelScope
-          scope.open(useInterruption).rethrow.flatMap { childScope =>
-            go(childScope, newExtendedScope, translation, boundToScope(childScope.id))
-          }
-        }
-        val frgx1 = interruptGuard(scope, view)(tail)
-        goView(frgx1, view)
+        val runner = new BuildR[G, X, End]
+        val tail = for {
+          closedExtendedScope <- maybeCloseExtendedScope
+          newExtendedScope = if (closedExtendedScope) None else extendedTopLevelScope
+          childScope <- scope.open(useInterruption).rethrow
+          bts = boundToScope(childScope.id)
+          fgrx <- go(childScope, newExtendedScope, translation, runner, bts)
+        } yield fgrx
```

Comment:
We don't have better-monadic-for on Dotty -- will this final suspended map cause a memory leak?

Reply:
Perhaps it could. I wrote these changes to make this code more readable, but they are not really necessary.